app/controllers/pensions/UpdatePensionProviderController.scala

```diff
@@ -81,20 +82,28 @@ trait UpdatePensionProviderController extends TaiBaseController
             case TaiSuccessResponseWithPayload(incomes: Seq[TaxCodeIncome]) =>
               incomes.find(income => income.employmentId.contains(id) &&
                 income.componentType == PensionIncome) match {
-                case Some(taxCodeIncome) =>
-                  journeyCacheService.cache(Map(UpdatePensionProvider_IdKey -> id.toString,
-                    UpdatePensionProvider_NameKey -> taxCodeIncome.name)).
-                    map {
-                      val model = PensionProviderViewModel(id, taxCodeIncome.name)
-                      _ => Ok(views.html.pensions.update.doYouGetThisPensionIncome(model, UpdateRemovePensionForm.form))
-                    }
+                case Some(taxCodeIncome) => cacheAndCreateView(id, taxCodeIncome)
                 case _ => throw new RuntimeException(s"Tax code income source is not available for id $id")
               }
             case _ => throw new RuntimeException("Tax code income source is not available")
           }
         }
   }
 
+  private def cacheAndCreateView(id: Int, taxCodeIncome: TaxCodeIncome)(implicit hc: HeaderCarrier,
+                                                                        request: Request[AnyContent],
+                                                                        user: TaiUser): Future[Result] = {
+    for {
+      _ <- journeyCacheService.cache(Map(UpdatePensionProvider_IdKey -> id.toString,
+        UpdatePensionProvider_NameKey -> taxCodeIncome.name))
+      optionalYesNo <- journeyCacheService.currentValue(UpdatePensionProvider_ReceivePensionQuestionKey)
```

Comment:
The `journeyCacheService.cache()` method will return the updated cache in full, so you could skip the extra hit on journey cache here - no need to go back and ask for currentValue.